### PR TITLE
Drop C bridge

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
The Python bridge is now in use from all of our downstreams which we release from main.  Let's drop the C bridge.

 - [x] Rebase/fix up after #21056

### CODE REVIEW

**Comment by allisonkarlitskaya** [comment]
I did a quick `grep`-based analysis of which files/functions in `src/common` are unused after dropping `src/bridge` and did some follow-ups.  I'm sure there's more stuff that we could find here if we were more thorough about it...

**Comment by allisonkarlitskaya** [comment]
The polkit thing needs a bit of thinking.  I have a sneaking suspicion that we actually need that file...

**Comment by martinpitt** [comment]
Yes, we do want the .policy file. It's got nothing to do with PCP, it's for cockpit-desktop (and perhaps Client if you use polkit instead of sudo)

**Comment by jelly** [comment]
```
msgfmt: cannot create output file "src/cockpit/data/org.cockpit-project.cockpit-bridge.policy": No such file or directory
make[2]: *** [Makefile:6019: src/cockpit/data/org.cockpit-project.cockpit-bridge.policy] Error 1
```

Hmm this is unexpected, does the unit test container not have a full checkout?

**Comment by jelly** [comment]
Oh, it does fail locally, it's lintian!

**Comment by jelly** [comment]
Debian should go green now, but more lintian warnings shown, we likely need to clean that up.

**Comment by martinpitt** [comment]
Looks great now! I retried the failed tests, pretty sure they were unrelated. But indeed the unit test failure needs fixing.

Please also squash the fixup commits -- unless we want to squash them all for the final landing?

**Comment by martinpitt** [comment]
> msgfmt: cannot create output file "src/cockpit/data/org.cockpit-project.cockpit-bridge.policy": No such file or directory

Could that be a srcdir vs. builddir issue? Should be reproducible locally with `make distcheck`.

File: src/Makefile.am [hunk 1]
```diff
@@ -86,3 +86,19 @@ nodist_metainfo_DATA = \
...
+
+# -----------------------------------------------------------------------------
+#  Policykit
+#
+polkit_in_files  = src/cockpit/data/org.cockpit-project.cockpit-bridge.policy.in
+EXTRA_DIST += $(polkit_in_files)
+
+if WITH_POLKIT
+polkitdir        = $(datadir)/polkit-1/actions
+polkit_DATA = $(polkit_in_files:.policy.in=.policy)
```
Comment (martinpitt) [inline src/Makefile.am:98]: The difference is that previously the .polkit.in and Makefile.am files were in the same dir, but this isn't the case any more. I bet it's confused by the builddir vs. srcdir path..

**Comment by jelly** [comment]
> Looks great now! I retried the failed tests, pretty sure they were unrelated. But indeed the unit test failure needs fixing.
> 
> Please also squash the fixup commits -- unless we want to squash them all for the final landing?

I don't want to step on @allisonkarlitskaya toes, just getting it green is the goal, then we should squash fixup commits in the relevant commits and :+1:

**Comment by jelly** [comment]
> > msgfmt: cannot create output file "src/cockpit/data/org.cockpit-project.cockpit-bridge.policy": No such file or directory
> 
> Could that be a srcdir vs. builddir issue? Should be reproducible locally with `make distcheck`.

```
if test -d "cockpit-325+git"; then find "cockpit-325+git" -type d ! -perm -700 -exec chmod u+rwx {} ';' ; rm -rf "cockpit-325+git" || { sleep 5 && rm -rf "cockpit-325+git"; }; else :; fi
make  \
  top_distdir="cockpit-325+git" distdir="cockpit-325+git" \
  dist-hook
make[4]: Entering directory '/home/jelle/projects/cockpit/20000/cockpit-325+git/_build/sub'
python3 '../..'/src/build_backend.py --copy '../..' 'cockpit-325+git'
Traceback (most recent call last):
  File "/home/jelle/projects/cockpit/20000/cockpit-325+git/_build/sub/../../src/build_backend.py", line 141, in <module>
    main()
  File "/home/jelle/projects/cockpit/20000/cockpit-325+git/_build/sub/../../src/build_backend.py", line 133, in main
    copy_sources(destdir)
  File "/home/jelle/projects/cockpit/20000/cockpit-325+git/_build/sub/../../src/build_backend.py", line 42, in copy_sources
    shutil.copy(source, destination)
  File "/usr/lib/python3.12/shutil.py", line 435, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/lib/python3.12/shutil.py", line 262, in copyfile
    with open(dst, 'wb') as fdst:
         ^^^^^^^^^^^^^^^
PermissionError: [Errno 13] Permission denied: '/home/jelle/projects/cockpit/20000/cockpit-325+git/_build/sub/cockpit-325+git/src/cockpit/data/org.cockpit-project.cockpit-bridge.policy.in'
```

So this is what I get locally with `distcheck`, seems we don't want policykit to be in src/cockpit/data

**Comment by jelly** [comment]
So this PR was green, but I messed up the autotools stuff. @allisonkarlitskaya care to take a look?

File: tools/debian/rules [hunk 2]
```diff
@@ -5,7 +5,7 @@ DEB_HOST_MULTIARCH ?= $(shell dpkg-architecture -qDEB_HOST_MULTIARCH)
 # we need an apparmor profile for >= 4.0; this exists in Ubuntu >= 24.04, but not in Debian yet
 PRE4AA = $(filter $(shell . /etc/os-release; echo $${VERSION_ID:-unstable}),22.04 11 12 unstable)
 
-CONFIG_OPTIONS := --disable-pcp
+CONFIG_OPTIONS :=
```
Comment (jelly) [inline tools/debian/rules:8]: This can be fully removed, I introduced this line and it is unneeded now. See 112ce31a06deee0d63cc255e6ce26cacfaeb2d9a

**Review by jelly — APPROVED** [review]
One tiny comment about Debian

**Comment by allisonkarlitskaya** [comment]
before and after:

image

Go TypeScript!  +0.3%!! :tada:

**Pull request merged**